drivers/network/dd/e1000/interrupt.c

```diff
@@ -44,12 +45,13 @@ MiniportHandleInterrupt(
     PE1000_ADAPTER Adapter = (PE1000_ADAPTER)MiniportAdapterContext;
     volatile PE1000_TRANSMIT_DESCRIPTOR TransmitDescriptor;
 
-    //NDIS_DbgPrint(MAX_TRACE, ("Called.\n"));
+    NDIS_DbgPrint(MAX_TRACE, ("Called.\n"));
 
     InterruptPending = InterlockedExchange(&Adapter->InterruptPending, 0);
 
-    E1000_LOCK_ADAPTER(Adapter);
+    // NdisDprAcquireSpinLock(&Adapter->AdapterLock);
```

Comment:
These functions don't modify the `Adapter` data?

Reply:
Link and Transmit parts change the data. TBH I don't know what to do with this lock

Reply:
I think we can safely drop all locks...